implementations/rust/ockam/ockam_identity/src/secure_channels/secure_channels.rs

```diff
@@ -1,60 +1,89 @@
-mod addresses;
-mod decryptor;
-mod decryptor_state;
-mod decryptor_worker;
-mod encryptor;
-mod encryptor_worker;
-mod listener;
-mod messages;
-
-mod common;
-mod local_info;
-mod options;
-mod registry;
-mod trust_policy;
-
-pub use common::*;
-pub use local_info::*;
-pub use options::*;
-pub use registry::*;
-pub use trust_policy::*;
-
-/// `AccessControl` implementation based on SecureChannel authentication guarantees
-pub mod access_control;
-/// SecureChannel API
-pub mod api;
-
-use crate::channel::addresses::Addresses;
-use crate::channel::decryptor_worker::DecryptorWorker;
-use crate::channel::listener::IdentityChannelListener;
-use crate::error::IdentityError;
-use crate::Identity;
+use crate::identities::Identities;
+use crate::identities::IdentitiesVault;
+use crate::identity::{Identity, IdentityError};
+use crate::secure_channel::{
+    Addresses, DecryptorWorker, IdentityChannelListener, Role, SecureChannelListenerOptions,
+    SecureChannelOptions, SecureChannelRegistry,
+};
+use crate::SecureChannelsBuilder;
 use core::time::Duration;
-use ockam_core::{Address, AsyncTryClone, Result, Route};
+use ockam_core::compat::sync::Arc;
+use ockam_core::Result;
+use ockam_core::{Address, Route};
+use ockam_node::Context;
+
+/// Identity implementation
+#[derive(Clone)]
+pub struct SecureChannels {
```

Comment:
Do you think we should wrap `SecureChannels` and other services inside an `Arc` if we can simply clone them instead?

Reply:
I did it mostly to stay flush with other services which were provided as `dyn TraitXxx` with the idea that we might extract an interface in the future.